board/technologic/common/micro/micro.c

```diff
@@ -0,0 +1,173 @@
+#include <stdint.h>
+#include <common.h>
+#include <asm/gpio.h>
+#include <linux/sizes.h>
+#include <asm/io.h>
+#include <common.h>
+#include <asm/arch/mx6-pins.h>
+#include <asm/mach-imx/iomux-v3.h>
+#include <i2c.h>
+#include <console.h>
+
+#include "micro.h"
+
+#define MIN_CHARGE_MV 3680
+#define MAX_CHARGE_MV 4800
+#define POWER_FAIL IMX_GPIO_NR(5, 0)
+#define MICRO_ADDR 0x54
+
+static bool read_power_fail_status(void);
+
+int micro_read(uint16_t addr, void *data, size_t size)
+{
+	struct udevice *dev;
+	int ret;
+
+	ret = i2c_get_chip_for_busnum(0, MICRO_ADDR, 2, &dev);
+	if (ret) {
+		pr_err("couldn't get i2c bus - %d\n", ret);
+		return ret;
+	}
+	return dm_i2c_read(dev, addr, data, size);
+}
+
+int micro_write(uint16_t addr, const void *data, size_t size)
+{
+	struct udevice *dev;
+	int ret = i2c_get_chip_for_busnum(0, MICRO_ADDR, 2, &dev);
+	if (ret) {
+		pr_err("couldn't get i2c bus - %d\n", ret);
+		return ret;
+	}
+
+	return dm_i2c_write(dev, addr, data, size);
+}
+
+int micro_read16_swap(int addr, uint16_t *data)
```

Comment:
This should be `static int` if its not intended to be available outside of this compilation unit.

Reply:
Fixed